src/core/server/plugins/discovery/plugin_manifest_parser.ts

```diff
@@ -160,6 +161,27 @@ export async function parseManifest(
     );
   }
 
+  if (
+    manifest.requiredOpenSearchPlugins !== undefined &&
+    !Array.isArray(manifest.requiredOpenSearchPlugins)
+  ) {
+    throw PluginDiscoveryError.invalidManifest(
+      manifestPath,
+      new Error(
+        `The "requiredOpenSearchPlugins" in plugin manifest for "${manifest.id}" should either be a string or an array of strings.`
```

Comment:
This error message doesn't match the `if` and will throw for strings. It also doesn't even check if the array is an array of strings.

Reply:
Ah good catch! Updated!